edx_when/api.py

```diff
@@ -357,6 +363,47 @@ def set_date_for_block(course_id, block_id, field, date_or_timedelta, user=None,
         return existing_date.id
 
 
+def get_schedules_with_due_date(course_id, assignment_date):
+    """
+    Get all Schedules with assignments due on a specific date for a Course.
+
+    Arguments:
+        course_id: either a CourseKey or string representation of same
+        assignment_date: a date object
+
+    Returns:
+        a QuerySet of Schedule objects for Users who have content due on the specified assignment_date
+    """
+    user_ids = models.UserDate.objects.select_related('content_date').select_related('content_date__policy').annotate(
+        computed_date=ExpressionWrapper(
+            F('content_date__policy__abs_date') + F('rel_date'),
+            output_field=DateTimeField()
+        ),
+    ).filter(
+        Q(computed_date__date=assignment_date, abs_date__isnull=True) |
+        Q(content_date__policy__abs_date__date=assignment_date, rel_date__isnull=True)
+    ).values_list('user_id', flat=True).distinct()
+
+    user_date_overridden_schedules = Schedule.objects.filter(
+        enrollment__course_id=course_id,
+        enrollment__user_id__in=user_ids,
+    )
+
+    # Get all relative dates for a course, we want them distinct, it doesn't matter how many of each due date there is
+    rel_dates = models.ContentDate.objects.filter(
+        course_id=course_id,
+    ).select_related('policy').values_list('policy__rel_date', flat=True).distinct()
+
+    # Using those relative dates, get all Schedules that have a "hit" by working backwards to the start_date
+    rel_start_dates = [assignment_date - rel_date for rel_date in rel_dates if rel_date is not None]
```

Comment:
I *think* you can do math using `F` expressions, to maybe leave this computation to the database: https://docs.djangoproject.com/en/3.0/ref/models/expressions/#supported-arithmetic

Reply:
Also, since you're filtering `None` out of the list in python, why not do it in the `rel_dates` query?

Reply:
> Also, since you're filtering `None` out of the list in python, why not do it in the `rel_dates` query?

👍 

> I _think_ you can do math using `F` expressions, to maybe leave this computation to the database: https://docs.djangoproject.com/en/3.0/ref/models/expressions/#supported-arithmetic

Ahh yeah, I did it above with 
```
computed_date=ExpressionWrapper(
    F('content_date__policy__abs_date') + F('rel_date'),
    output_field=DateTimeField()
),
``` 
The issue I ran into below was that I wasn't querying with any values from the Schedule table, I already knew what values I needed to look for, and I wasn't positive how to sort of work the other way if that makes sense. Also this seemed reasonable enough, and I don't think we'll ever run into the problem of having too many relative dates for a single course where an `in` query wouldn't be sufficient.

Happy to do an approach more like ```    .annotate(
        has_assignment_due=ExpressionWrapper(
            assignment_date - F('start_date__date') in rel_dates,
            BooleanField()
        ),
    ).filter(has_assignment_due=True)``` if you think that's more readable?

Reply:
My main concern was more about just keeping as much work in the database and out of application code as possibe.